<launch>

<!-- OBJECT RECOGNIZER DEMO -->
	 <node name="qbo_stereo_selector" pkg="qbo_stereo_selector" type="qbo_stereo_selector" >
          <param name="distance_filter_threshold" value="0.53" />

	 </node>
 	 <param name="/qbo_stereo_selector/move_head" value="false"/> 

	 <node name="orbit_client" pkg="qbo_object_recognition" type="orbit_client" />
          <param name="/qbo_object_recognition/input_image_topic" value="/qbo_stereo_selector/object"/> 
	 
         <!-- <node name="orbit_demo" pkg="qbo_object_recognition" type="orbit_demo"/> -->
         <node name="orbit_demo_lang" pkg="qbo_object_recognition" type="orbit_demo"/>
	 
<!-- FACE RECOGNIZER DEMO -->


	<!-- Face Tracking node to detect faces -->
	<node name="qbo_face_tracking" pkg="qbo_face_tracking" type="qbo_face_tracking" >
	 </node>
	 <!-- <param name="/qbo_face_tracking/face_classifier_path" value="0" /> -->
	 <!-- <param name="/qbo_face_tracking/default_pos_x" value="0" /> -->
	 <!-- <param name="/qbo_face_tracking/default_pos_y" value="10" /> -->
	  <param name="/qbo_face_tracking/check_Haar" value="50" /> 
	 <!-- <param name="/qbo_face_tracking/check_track_object" value="50" /> -->
	 <!-- <param name="/qbo_face_tracking/undetected_threshold" value="5.0" /> -->
	 <!-- <param name="/qbo_face_tracking/distance_threshold" value="5.0" /> -->
	 <param name="/qbo_face_tracking/send_to_recognizer" value="true" /> 
	 <param name="/qbo_face_tracking/print_recognized_face" value="false" />
	
	<!-- Face Following node to move head and base -->
	<node name="qbo_face_following" pkg="qbo_face_following" type="qbo_face_following">
	 </node> 
	 <param name="/qbo_face_following/move_base" value="false" />
	 <param name="/qbo_face_following/move_head" value="true" />
	 <param name="/qbo_face_following/search_min_pan" value="-0.7" /> 
	 <param name="/qbo_face_following/search_max_pan" value="0.7" /> 
	 <param name="/qbo_face_following/search_pan_vel" value="0.3" />  
	 <param name="/qbo_face_following/search_min_tilt" value="-0.2" /> 
	 <param name="/qbo_face_following/search_max_tilt" value="0.0" />
     <param name="/qbo_face_following/search_tilt_vel" value="0.3" /> 
     
     <param name="/qbo_face_following/desired_distance" value="1.5" /> 
	
	 <node name="qbo_face_recognition" pkg="qbo_face_recognition" type="qbo_face_recognition" >
	 </node>
	<!-- <param name="/qbo_face_recognition/recognition_type" value="1" /> -->       
	 <param name="/qbo_face_recognition/stabilizer_threshold" value="16" />
	 <param name="/qbo_face_recognition/stabilizer_max" value="20" /> 
	<!-- <param name="/qbo_face_recognition/num_of_desc_per_face_" value="50" /> -->
	 <param name="/qbo_face_recognition/bow_certainty_threshold" value="0.25" /> 
	<!-- <param name="/qbo_face_recognition/descriptors_match_threshold" value="0.23" /> -->  
	<!-- <param name="/qbo_face_recognition/linear_kernel" value="false" /> -->      
	<!-- <param name="/qbo_face_recognition/pca_dimension" value="40" /> -->  
	<!-- <param name="/qbo_face_recognition/pca_image_height" value="100" /> -->
	<!-- <param name="/qbo_face_recognition/pca_image_width" value="100" /> -->
	
	<!-- Face Recognition Demo to answer questions and to learn faces -->
	<node name="qbo_face_recognition_demo" pkg="qbo_face_recognition" type="face_recognizer_demo" output="screen" ></node>

	 <param name="/qbo_face_recognition_demo/num_images_to_hold" value="20" />
	 <param name="/qbo_face_recognition_demo/wait_for_name_tolerance" value="4.0" />

<!-- QBO CHATTER -->
	<!-- <node name="qbo_questions" pkg="qbo_questions" type="questions.py" /> -->
	<!-- <node name="qbo_system_info" pkg="qbo_system_info" type="plugin_system.py" /> -->
        <!-- <node name="qbo_expression" pkg="qbo_expression" type="qbo_expression.py" /> -->

</launch>

